app/upgrades/v15/upgrade_test.go

```diff
@@ -67,6 +77,62 @@ func (suite *UpgradeTestSuite) TestMigrateNextPoolIdAndCreatePool() {
 	suite.Require().Equal(gammPoolCreationFee, poolmanagerPoolCreationFee)
 }
 
+
+func (suite *UpgradeTestSuite) TestMigrateBalancerToStablePools() {
+	suite.SetupTest() // reset
+
+	ctx := suite.Ctx
+	gammKeeper := suite.App.GAMMKeeper
+	poolmanagerKeeper := suite.App.PoolManagerKeeper
+	// bankKeeper := suite.App.BankKeeper
+	testAccount := suite.TestAccs[0]
+
+	// Mint some assets to the accounts.
+	suite.FundAcc(testAccount, DefaultAcctFunds)
+
+	// Create the balancer pool
+	swapFee, err := sdk.NewDecFromStr("0.003")
+	exitFee, err := sdk.NewDecFromStr("0.025")
+	poolID, err := suite.App.PoolManagerKeeper.CreatePool(
+		suite.Ctx,
+		balancer.NewMsgCreateBalancerPool(suite.TestAccs[0],
+		balancer.PoolParams{
+			SwapFee: swapFee,
+			ExitFee: exitFee,
+		},
+		[]balancertypes.PoolAsset{
+			{
+				Weight: sdk.NewInt(100),
+				Token:  sdk.NewCoin("foo", sdk.NewInt(5000000)),
+			},
+			{
+				Weight: sdk.NewInt(200),
+				Token:  sdk.NewCoin("bar", sdk.NewInt(5000000)),
+			},
+		},
+		""),
+	)
+	_ = poolID
+	suite.Require().NoError(err)
+
+	// create the pool
+	balancerPool, err := gammKeeper.GetPool(suite.Ctx, poolID)
+	suite.Require().NoError(err)
+	balancerShares := balancerPool.GetTotalShares()
+
+	// test migrating the balancer pool to a stable pool
+	v15.MigrateBalancerPoolToSolidlyStable(ctx, gammKeeper, poolmanagerKeeper, poolID)
+
+	// check that the pool is now a stable pool
+	stablepool, err := gammKeeper.GetPool(ctx, poolID)
+	suite.Require().NoError(err)
+	suite.Require().Equal(stablepool.GetType(), poolmanagertypes.Stableswap)
+
+	// check that the number of stableswap LP shares is the same as the number of balancer LP shares
+	suite.Require().Equal(balancerShares.String(), stablepool.GetTotalShares().String())
```

Comment:
@p0mvn per your comment

> The upgrade handler unit test can help assert that the stable swap pool was correctly initialized with the desired liquidity.

Is this check sufficient?

Reply:
Let's also add a check for [liquidity](https://github.com/osmosis-labs/osmosis/blob/a4e48c6d12620f64dc0c114985650e9544cd00c5/x/gamm/pool-models/stableswap/pool.go#L100-L104) and bank balances please

Reply:
What do you mean by bank balances?

Reply:
If this is related to all addresses holdings of LP tokens, might need a similar function as referenced here https://github.com/osmosis-labs/osmosis/pull/4384/files#r1114946236

Reply:
I mean both the LP tokens and token liquidity. For example, `stOsmo` and `osmo`. Pool addresses have both

Reply:
Done.